src/lake/Lake/DSL/Meta.lean

```diff
@@ -53,9 +56,42 @@ scoped syntax (name := metaIf)
 "meta " "if " term " then " cmdDo (" else " cmdDo)? : command
 
 elab_rules : command | `(meta if $c then $t $[else $e?]?) => do
-  if (← withRef c <| runTermElabM fun _ => evalTerm Bool c) then
+  if (← withRef c <| runTermElabM fun _ => evalTerm Bool (toTypeExpr Bool) c .unsafe) then
     let cmd := mkNullNode (expandCmdDo t)
     withMacroExpansion (← getRef) cmd <| elabCommand cmd
   else if let some e := e? then
     let cmd := mkNullNode (expandCmdDo e)
     withMacroExpansion (← getRef) cmd <| elabCommand cmd
+
+@[implemented_by Meta.evalExpr]
+opaque evalExpr (α) (expectedType : Expr) (value : Expr) (safety := DefinitionSafety.safe) : MetaM α
+
+def toExprIO [ToExpr α] (x : IO α) : IO Expr :=
+  toExpr <$> x
+
+/--
+Executes a term of type `IO α` at elaboration-time
+and produces an expression corresponding to the result via `ToExpr α`.
+-/
+scoped syntax:lead (name := runIO) "run_io " term : term
```

Comment:
```suggestion
scoped syntax:lead (name := runIO) "run_io " doSeq : term
```
It is generally more convenient to have monad-taking elabs directly receive a `doSeq` so you don't need to do `run_io do` and `<-` works as intended.